### PR TITLE
switcher for preconditioner

### PR DESCRIPTION
**Pull request by ESeNonFossiIo**
This PR allows to switch to a preconditioner assemble with or without back_forward_substitution.

### CODE REVIEW

File: include/interfaces/stokes.h [hunk 1]
```diff
@@ -211,26 +213,42 @@ Stokes<dim>::compute_system_operators(const DoFHandler<dim> &dh,
...
+        }
+      });
 
-  // ASSEMBLE THE PROBLEM:
-  system_op  = block_operator<2, 2, VEC >({{
-      {{ A, Bt }} ,
-      {{ B, ZeroP }}
-    }
-  });
+      const auto DiagInv = block_operator<2, 2, VEC >({{
```
Comment (ESeNonFossiIo) [inline include/interfaces/stokes.h:224]: non riesco ad usare block_diagonal_operator perché ho errori di questo tipo:

```
/Applications/deal.II.app/Contents/Resources/include/deal.II/lac/trilinos_parallel_block_vector.h:502:25: error: no member named 'range_partitioner' in 'dealii::LinearOperator<dealii::TrilinosWrappers::MPI::Vector, dealii::TrilinosWrappers::MPI::Vector>'
        v.reinit(matrix.range_partitioner(), fast);
```

**Comment by ESeNonFossiIo** [comment]
[PR #18](https://github.com/mathLab/pi-DoMUS/issues/18) 
Check that the current preconditioner for Stokes works with Linear Operators

**Comment by luca-heltai** [comment]
Can you provide a minimal failing test? I think there might be a bug in the diagonal operator, and we should be able to fix it quite easily by manually creating a working alternative to the diagonal operator (pr to deal.II?)

**Comment by ESeNonFossiIo** [comment]
I tried to provide a minimal test but every test based on the tests of deal.ii passes.
I really do not know how to highlight this bug.

**Pull request merged**